### PR TITLE
[Merged by Bors] - feat(Topology/KrullDimension): add invariance under homeomorphism theorem and auxiliary lemmas

### PR DESCRIPTION
**Pull request by Raph-DG**
Added some lemmas about the topological Krull dimension

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Set/Image.lean [hunk 1]
```diff
@@ -1090,6 +1090,10 @@ theorem Injective.image_injective (hf : Injective f) : Injective (image f) := by
   intro s t h
   rw [← preimage_image_eq s hf, ← preimage_image_eq t hf, h]
 
+lemma Injective.image_strictMono {X Y : Type*}
```
Comment (riccardobrasca) [inline Mathlib/Data/Set/Image.lean:1093]: I guess there is an `f` already in scope here.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

**Comment by Raph-DG** [comment]
Thanks for the reviews!

bors r+

**Pull request closed without merging**